### PR TITLE
Bugfix FXIOS-8309 [v122.1] System theme resetting bug

### PR DESCRIPTION
**Pull request by adudenamedruby**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8309)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18421)

## :bulb: Description
the migration was resetting the system theme.

Also, I started removing a bit more legacy theme stuff. Just a bit.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

**Review by lmarceau — COMMENTED** [review]
Will this PR conflicts with the legacy removal PR?

**Comment by adudenamedruby** [comment]
It won't, because that legacy one was based off this. So this needs to be merged first, then that. :)

**Comment by adudenamedruby** [comment]
@mergifiyo backport release/v122 release/v123

**Review by lmarceau — APPROVED** [review]

File: firefox-ios/Client/Frontend/Settings/ThemeSettings/ThemeMiddleware.swift [hunk 1]
```diff
@@ -79,10 +78,10 @@ class ThemeManagerMiddleware: ThemeManagerProvider {
         themeManager.setAutomaticBrightness(isOn: enabled)
     }
 
-    func updateManualTheme(_ theme: ThemeType) {
-        let isLightTheme = theme == .light
+    func updateManualTheme(_ newTheme: ThemeType) {
+        let isLightTheme = newTheme == .light
         legacyThemeManager.current = isLightTheme ? LegacyNormalTheme() : LegacyDarkTheme()
-        themeManager.changeCurrentTheme(isLightTheme ? .light : .dark)
+        themeManager.changeCurrentTheme(newTheme)
```
Comment (cyndichin) [inline firefox-ios/Client/Frontend/Settings/ThemeSettings/ThemeMiddleware.swift:84]: logic changed here in which we don't default to dark theme, are we okay with this update?

**Review by cyndichin — APPROVED** [review]
thanks for the update! looks a lot cleaner as well :fire: 
one comment for double checking, not a blocker.

File: firefox-ios/Client/Frontend/Settings/ThemeSettings/ThemeMiddleware.swift (hunk 1 above)
Comment (adudenamedruby) [reply firefox-ios/Client/Frontend/Settings/ThemeSettings/ThemeMiddleware.swift:84]: Well, this should default to whatever's called, really, as opposed to having a random default. It's cleaner this way and the logic is easier to follow, imo.

**Comment by adudenamedruby** [comment]
[BR is green](https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/32464559-5524-4d92-b2b1-6e9ecd7692a2)

**Pull request merged**